specs/NotifyDropTargetAction.md

```diff
@@ -0,0 +1,280 @@
+    in C++ for our COM API. Use snippets of the sample code you wrote for the sample apps.
+
+    The general format is:
+
+    ## FirstFeatureName
+
+    Feature explanation text goes here, including why an app would use it, how it
+    replaces or supplements existing functionality.
+
+    ```c#
+    void SampleMethod()
+    {
+        var show = new AnExampleOf();
+        show.SomeMembers = AndWhyItMight(be, interesting)
+    }
+    ```
+    
+    ```cpp
+    void SampleClass::SampleMethod()
+    {
+        winrt::com_ptr<ICoreWebView2> webview2 = ...
+    }
+    ```
+
+    ## SecondFeatureName
+
+    Feature explanation text goes here, including why an app would use it, how it
+    replaces or supplements existing functionality.
+
+    ```c#
+    void SampleMethod()
+    {
+        var show = new AnExampleOf();
+        show.SomeMembers = AndWhyItMight(be, interesting)
+    }
+    ```
+    
+    ```cpp
+    void SampleClass::SampleMethod()
+    {
+        winrt::com_ptr<ICoreWebView2> webview2 = ...
+    }
+    ```
+
+    As an example of this section, see the Examples section for the PasswordBox
+    control (https://docs.microsoft.com/windows/uwp/design/controls-and-patterns/password-box#examples). 
+-->
+
+```c++
+// Initialized elsewhere
+wil::com_ptr<ICoreWebView2Controller> webViewController;
+
+webViewCompositionController =
+        webViewController.query<ICoreWebView2CompositionController>();
+
+// Implementation for IDropTarget
+HRESULT DropTarget::DragEnter(IDataObject * data_object,
+    DWORD key_state,
+    POINTL cursor_position,
+    DWORD* effect)
+{
+    POINT point = {cursor_position.x, cursor_position.y};
+    // Tell the helper that we entered so it can update the drag image and return
+    // the allowed effects.
+    IDropTargetHelper* drop_helper = DropHelper();
+    if (drop_helper) {
+        drop_helper->DragEnter(GetHWND(), data_object, &point, *effect);
+    }
+    // Convert the point to WebView client coordinates from screen coordinates
+    // account for any offset.
+    ConvertScreenPointToWebView(&point);
+    return webViewCompositionController->NotifyDropTargetAction(
+        COREWEBVIEW2_DRAG_ENTER, data_object, key_state, point, effect);
+}
+
+HRESULT DropTarget::DragOver(DWORD key_state,
+    POINTL cursor_position,
+    DWORD * effect)
+{
+    POINT point = {cursor_position.x, cursor_position.y};
+    IDropTargetHelper* drop_helper = DropHelper();
+    if (drop_helper)
+        drop_helper->DragOver(&point, *effect);
+    ConvertScreenPointToWebView(&point);
+    return webViewCompositionController->NotifyDropTargetAction(
+        COREWEBVIEW2_DRAG_OVER, nullptr, key_state, point, effect);
+}
+
+HRESULT DropTarget::DragLeave()
+{
+    IDropTargetHelper* drop_helper = DropHelper();
+    if (drop_helper)
+        drop_helper->DragLeave();
+    return webViewCompositionController->NotifyDropTargetAction(
+        COREWEBVIEW2_DRAG_LEAVE, nullptr, 0, {0, 0}, nullptr);
```

Comment:
It seems like the different actions have different parameters which makes me think these should be different methods rather than one with different enum values.

Is there a reason there's just one method with an enum to pick which underlying method is getting called? Based on your sample here it doesn't seem to make it easier for the end developer to have one method and different enum values? If so, we need to somehow make this evident in the API spec. If not, then we should have separate methods =)